src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -427,6 +424,38 @@ internal static SecurityKey ResolveTokenSigningKey(string kid, string x5t, Token
 
             return null;
         }
+
+        /// <summary>
+        /// Returns number of Jwt parts in token.
+        /// </summary>
+        /// <param name="token">Input token to count patrs.</param>
+        /// <returns>Value between 1 and <see cref="JwtConstants.MaxJwtSegmentCount"/> + 1.</returns>
+        internal static int CountTokenJwtParts(string token)
+        {
+            int tokenPartsCount = 1;
+            for (int i = 0; i < token.Length; i++)
+            {
+                if (token[i] == '.')
+                {
+                    tokenPartsCount++;
+                    if (tokenPartsCount > JwtConstants.MaxJwtSegmentCount)
```

Comment:
We need to fault here if the number of dots doesn't equal 3 or 5.

Reply:
@brentschmaltz the method cannot throw an exception because it is used in method CanReadToken that returns false when an incorrect token is provided.

Reply:
That is a good point. 
I agree.

Reply:
@lechu445 i'll run some numbers on it and we should get this in.